### PR TITLE
#332 add project omada controller as rock on

### PR DESCRIPTION
**Pull request by janholtzhausen**
Basics done (I think), please feel free to correct me for future rock-ons :)

### CODE REVIEW

**Comment by Hooverdan96** [comment]
Fixes #332

File: omada-controller.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                        "label": "EAPController LOGS Storage"
+                    }
+                },
+                "environment": {
+                    "TZ": {
+                    "description": "Timezone for the controller.",
+                    "label": "TZ",
+                    "index": 1	
+                    }
+                }
```
Comment (Hooverdan96) [inline omada-controller.json:64]: The timezone mapping is not really required. The Rockon helper already maps those to the TZ of the Rockstor appliance
```suggestion

```

**Comment by Hooverdan96** [comment]
@janholtzhausen, I believe the Timzone mapping is not required at this time, as Rockstor always includes it as part of the Docker command it issues, tied to the appliance timezone.

I noticed, that the container changes the permissions on the share to the default process ID/group (508/508). Would it be better to expose the PUID/PGID parameters in the configuration, so one has better control over on how the shares are managed, as opposed to the Rockon managing the share (and one then not necessarily having access to it anymore, unless using the root user)?

Since I don't have relevant TP-Link controllers, I can't really test, but does this setup require the `net=host` setting? I wasn't sure, and we are striving to not have to use it, but just wanted to make sure, that it will work without it.

File: omada-controller.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                    },
+                    "/opt/tplink/EAPController/work": {
+                        "description": "Choose a WORK Share for Omada Controller files. Eg: create a Share called omada-controller-work for this purpose alone.",
+                        "label": "EAPController WORK Storage"
+                    },
+                    "/opt/tplink/EAPController/logs": {
+                        "description": "Choose a LOGS Share for Omada Controller files. Eg: create a Share called omada-controller-logs for this purpose alone.",
+                        "label": "EAPController LOGS Storage"
+                    }
+                },
```
Comment (Hooverdan96) [inline omada-controller.json:57]: @janholtzhausen. My apologies, as part of the changes requested I overlooked that there was a superfluous comma hanging around. Could you remove that one (it then passes the JSON lint test, too).
```suggestion
                }
```

**Comment by Hooverdan96** [comment]
@janholtzhausen thank you! I'll test one more time tomorrow and then it should be good to go from my perspective.

**Comment by janholtzhausen** [comment]
Hi,
I think it;s necessary, since the protocols are not too different from the Ubiquity kit.

**Comment by Hooverdan96** [comment]
@janholtzhausen, thinking about it some more, if the current version you pushed here works for you (meaning running the container within the Rockon framework and connecting to the Omada controller is not a problem for you) then it does not seem to be necessary. And would be what we'd prefer anyway, since it makes port management simpler between Rockons.
Otherwise, if you could update the json file to contain the parameter variable?
image

**Comment by kanecko** [comment]
I'm amidst creating my own Rockon, because I have discovered this PR too late.
What's the progress on this PR? Was it canceled?

**Comment by phillxnet** [comment]
@kanecko Hello there. Re:

> What's the progress on this PR? Was it canceled?

I'm not actually sure myself. From the pr history it looks like @Hooverdan96 was exploring the need for the --net=host option as we only like to accept that if it's absolutely necessary, given it gives the docker image full network access. There seems to have been a question as to it's need or otherwise.

A month later @janholtzhausen looks to have closed the pull request. It could be that they lost interest in the submission or no longer had access to a rockstor system to prove any changes.

If you fancy submitting your own then do go ahead, we are limited in the number of folks who review these Rock-on submissions however so a bit of patience may be required unfortunately. Just take a looks at our other Rock-ons and some more recent accepted (merged) Rock-on submissions to see the type and format of information as that should drastically speed up the review process. The actual publishing is relatively trivial however.

**Comment by kanecko** [comment]
Sure, why not 😊
Will create a new PR at some point.

**Comment by janholtzhausen** [comment]
Afraid my Rockstor solution became redundant.

 

I was pretty sure that host level access was still required, just because of an element of dynamic port usage for device discovery (pretty much the same situation as the Ubiqity controller rock-on, which this was shamelessly copied from…)

 

I can spin up a virtual rockstor to re-test, but I’m afraid a nearly yearlong delay between PR and acceptance may be too long for me to keep a dev instance alive in the lab.

 

 

Regards

Jan

**Comment by kanecko** [comment]
Thanks for clarifying!
I have my own Omada Rockon up and running without the `--net=host` option. At least the WebUI works.
I will set up my Omada devices next, to see, if it still works without it.

**Comment by kanecko** [comment]
Everything is working fine.
submitted PR #341

**Pull request closed without merging**